hypershift-operator/controllers/hostedcluster/internal/platform/powervs/powervs.go

```diff
@@ -326,6 +326,40 @@ func (p PowerVS) ReconcileCredentials(ctx context.Context, c client.Client, crea
 	if err != nil {
 		return fmt.Errorf("failed to reconcile storage operator provider creds: %w", err)
 	}
+
+	// Reconcile the platform provider image registry operator credentials secret by
+	// resolving  the reference from the HostedCluster and syncing the secret in
+	// the control plane namespace.
+	if err = c.Get(ctx, client.ObjectKey{Namespace: hcluster.GetNamespace(), Name: hcluster.Spec.Platform.PowerVS.ImageRegistryOperatorCloudCreds.Name}, &src); err != nil {
+		return fmt.Errorf("failed to get image registry operator provider creds %s: %w", hcluster.Spec.Platform.PowerVS.ImageRegistryOperatorCloudCreds.Name, err)
+	}
+	dest = &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: controlPlaneNamespace,
+			Name:      src.Name,
+		},
+	}
+	if _, err = createOrUpdate(ctx, c, dest, func() error {
```

Comment:
I think you'd want to validate the expectations for this key in the hc controller config validation so a failure is signalled in a condition, otherwise this will silently fail.

Reply:
Thanks for the input, will take that as a separate issue and address it.